assume/units/dst_components.py

```diff
@@ -0,0 +1,903 @@
+        self.b.storage_loss_rate_dri = pyo.Param(initialize=self.storage_loss_rate)
+        self.b.charge_loss_rate_dri = pyo.Param(initialize=self.charge_loss_rate)
+        self.b.discharge_loss_rate_dri = pyo.Param(initialize=self.discharge_loss_rate)
+
+    def define_variables(self, time_steps):
+        self.b.soc_dri = pyo.Var(time_steps, within=pyo.NonNegativeReals)
+        self.b.uniformity_indicator_dri = pyo.Var(time_steps, within=pyo.Binary)
+
+        # Define the variables for power and hydrogen
+        self.b.charge_dri = pyo.Var(time_steps, within=pyo.NonNegativeReals)
+        self.b.discharge_dri = pyo.Var(time_steps, within=pyo.NonNegativeReals)
+
+    def define_constraints(self, time_steps):
+        """
+        Ensures the SOC of the DRI storage unit stays above the minimum capacity.
+        """
+
+        @self.b.Constraint(time_steps)
+        def storage_min_capacity_dri_constraint(b, t):
+            return b.soc_dri[t] >= b.min_capacity_dri
+
+        """
+        Ensures the SOC of the DRI storage unit stays below the maximum capacity.
+        """
+
+        @self.b.Constraint(time_steps)
+        def storage_max_capacity_dri_constraint(b, t):
+            return b.soc_dri[t] <= b.max_capacity_dri
+
+        """
+        Limits the charging of the DRI storage unit to its maximum capacity.
+        """
+
+        @self.b.Constraint(time_steps)
+        def energy_in_max_capacity_dri_constraint(b, t):
+            return b.charge_dri[t] <= b.max_capacity_dri
+
+        """
+        Limits the discharging of the DRI storage unit to its maximum capacity.
+        """
+
+        @self.b.Constraint(time_steps)
+        def energy_out_max_capacity_dri_constraint(b, t):
+            return b.discharge_dri[t] <= b.max_capacity_dri
+
+        """
+        Ensures uniformity in charging the DRI storage unit.
+        """
+
+        @self.b.Constraint(time_steps)
+        def energy_in_uniformity_dri_constraint(b, t):
+            return b.charge_dri[t] <= b.max_capacity_dri * b.uniformity_indicator_dri[t]
+
+        """
+        Ensures uniformity in discharging the DRI storage unit.
+        """
+
+        @self.b.Constraint(time_steps)
+        def energy_out_uniformity_dri_constraint(b, t):
+            return b.discharge_dri[t] <= b.max_capacity_dri * (
+                1 - b.uniformity_indicator_dri[t]
+            )
+
+        """
+        Defines the change in SOC of the DRI storage unit over time.
+        """
+
+        @self.b.Constraint(time_steps)
+        def storage_capacity_change_dri_constraint(b, t):
+            return b.soc_dri[t] == (
+                (
+                    (b.soc_dri[t - 1] if t > 0 else b.initial_soc_dri)
+                    * (1 - b.storage_loss_rate_dri)
+                )
+                + ((1 - b.charge_loss_rate_dri) * b.charge_dri[t])
+                - ((1 + b.discharge_loss_rate_dri) * b.discharge_dri[t])
+            )
```

Comment:
It seems that these "classes" do not have any kind of state or methods at all.
They could be modeled as well as factory functions, which receive a configuration dict and return the created pyomo block.

so that in steel_plant one simplyhas:

`self.model.add_block(create_storage_unit(**component_data))`

I would like that much better, except you see possibility that these classes will do much more in the future.